src/05-derivatives/05-diffusion-derivatives.md

```diff
@@ -18,132 +18,359 @@ should be included in the pipeline documentation.
+| GradientNonLinearityGeometryCorrection | OPTIONAL. Boolean. Correction for geometric distortions arising from non-linearity of gradients                                                   |
+| GradientNonLinearityQSpaceCorrection   | OPTIONAL. Boolean. Correction for spatial inhomogeneity of diffusion sensitisation gradient strength                                              |
+| SliceDropoutDetection                  | OPTIONAL. Boolean. Detection of signal dropout in acquired slices during pre-processing                                                           |
+| SliceDropoutReplacement                | OPTIONAL. Boolean. Replacement of image data within slices containing signal dropout with predicted values                                        |
+| BiasFieldCorrectionMethod              | OPTIONAL. String. Method (if any) used for correction of B1 RF field inhomogeneity                                                                |
 
-## Diffusion Models (and input parameters)
+## Diffusion models
 
-Diffusion MRI can be modeled using various paradigms to extract a more easily
-understandable representation of the diffusion process and the underlying
-structure. To do so, parameters might be needed to control how the signal is fit
-to the model. Those parameters are called **input parameters** in the following.
-Once the model is fit, the resulting representation can be saved using a number
-of values per voxel. Those values will be called **output** or **estimated
-parameters** in the following.
+Diffusion MRI can be modeled using various paradigms to extract more
+informative representations of the diffusion process and the underlying
+biological structure. A wide range of such models are available, each of
+which has its own unique requirements with respect to:
 
-**Estimated parameters** are saved as NIFTI files (see section Models for the
-expected content of each model), and **input parameters** are saved in the
-sidecar JSON file.
+-   The input parameters required to define / constrain the model;
 
-The following is a general example of naming convention:
+-   The appropriate data representation in which the resulting
+    parameters estimated of / from the model are stored on the filesystem;
+
+-   The requirements for encapsulation and complete representation of
+    derived orientation information, which is a key strength of diffusion
+    MRI but presents unique challenges for correct interpretation.
+
+### Parameter terminology
+
+1.  *Intrinsic* parameter: Item that is the direct result of fitting
+    the diffusion model to the empirical diffusion-weighted data.
+
+1.  *Input* parameter: Item that influences the conformation of the
+    diffusion model to the empirical diffusion-weighted data.
+
+1.  *Extrinsic* parameter: Item that can be calculated from prior

```

Comment:
Every time I tried to more stringently separate intrinsic and extrinsic parameters it always felt artificial. Even though they're *derived* differently, their actual representation on disk, and potential subsequent utility, isn't actually really that much affected by whether or not "the DWI data were being referred to in the generation of the parameter". So there would end up being two different definitions and file naming conventions for two pieces of data that were otherwise "equals". It can even get tricky whereby a particular software tool can export a parameter as part of the fitting procedure that's actually extrinsic, but was generated and saved along with intrinsic parameters, either out of convenience or for a lack of dedicated stand-alone command for generating such or even for simply not being interested in the distinction.

I don't see that fitting a model with one tool and extracting extrinsic parameters with another is incompatible with the current proposal. There's one JSON across all intrinsic model parameters that can contain the software used to fit the model, but each extrinsic parameter can additionally have its own JSON, which could contain the software used to extract that particular parameter if desired.

I feel like the "modality" as you put it, ie. the final key in the file name (is that a common nomenclature for it?), should have the minimal cardinality possible across BIDS. Adding one *per parameter* across all possible diffusion model parameters would very much affect that.

Reply:
Following-up here #12.